src/twisted/protocols/tls.py

```diff
@@ -196,7 +198,9 @@ def makeConnection(self, transport):
         Connect this wrapper to the given transport and initialize the
         necessary L{OpenSSL.SSL.Connection} with a memory BIO.
         """
-        self._tlsConnection = self.factory._createConnection(self)
+        # Using weakref.proxy to avoid circular references
+        selfProxy = weakref.proxy(self)
```

Comment:
just asking. Is using a weakref the only way solve this?

Why not break it in `connectionLost` like done in the other place?

-------

Using weakref can make the tests harder to read / write as you will need to take extra care on the scope of an instance..

...and I would like to write Python without having to care about the lifecycle of the objects.